test/thermo/CoverageDependentSurfPhase_Test.cpp

```diff
@@ -0,0 +1,437 @@
+#include "gtest/gtest.h"
+#include "cantera/thermo/SurfPhase.h"
+#include "cantera/base/utilities.h"
+#include "cantera/thermo/CoverageDependentSurfPhase.h"
+#include "cantera/thermo/ThermoFactory.h"
+
+
+namespace Cantera
+{
+
+class CoverageDependentSurfPhase_Test: public testing::Test
+{
+public:
+    CoverageDependentSurfPhase_Test():
+        // Array of ten different temperatures to test if
+        // temperature dependence is implemented properly
+        test_Ts({500.0, 625.0, 427.0, 711.0, 260.0,
+                 903.0, 475.3, 748.3, 100.1, 372.1}),
+        // Array of ten different surface coverages to test if
+        // coverage dependence is implemented properly
+        test_covs({
+                  {0.00, 0.10, 0.50, 0.10, 0.30},
+                  {0.12, 0.07, 0.21, 0.17, 0.43},
+                  {0.00, 0.71, 0.08, 0.07, 0.14},
+                  {0.60, 0.10, 0.10, 0.10, 0.10},
+                  {0.20, 0.23, 0.57, 0.00, 0.00}
+                  })
+    {
+        // CoverageDenpendentSurfPhase method values will be
+        // compared against SurfPhase method values or explicitly calculated values
+        covdepsurf_phase.reset(newPhase("copt_covdepsurf_example.yaml", "covdep"));
+        idealsurf_phase.reset(newPhase("copt_covdepsurf_example.yaml", "ideal"));
+        surfphase_ptr = dynamic_cast<SurfPhase*>(covdepsurf_phase.get());
+    }
+    std::unique_ptr<ThermoPhase> covdepsurf_phase;
+    std::unique_ptr<ThermoPhase> idealsurf_phase;
+    // A pointer to call setCoverage method
+    SurfPhase* surfphase_ptr;
+    // To call unit convert methods
+    UnitSystem us;
+
+    vector_fp test_Ts;
+    std::vector<vector_fp> test_covs;
+};
+
+TEST_F(CoverageDependentSurfPhase_Test, construct_from_yaml)
+{
+    // This test checks if CoverageDependentSurfPhase object is
+    // properly created without errors
+    CoverageDependentSurfPhase* coveragedependentsurf_phase =
+        dynamic_cast<CoverageDependentSurfPhase*>(covdepsurf_phase.get());
+    EXPECT_TRUE(coveragedependentsurf_phase != NULL);
```

Comment:
I'd suggest that this particular test is unnecessary - if the phase isn't actually instantiated, there will be plenty of other tests that fail as well. (I know it's following a pattern from some of the other thermo tests, but those are on their way out).